internal/pkg/agent/cmd/watch.go

```diff
@@ -93,14 +102,18 @@ func watchCmd(log *logp.Logger, cfg *configuration.Configuration) error {
 		_ = locker.Unlock()
 	}()
 
-	isWithinGrace, tilGrace := gracePeriod(marker, cfg.Settings.Upgrade.Watcher.GracePeriod)
-	if !isWithinGrace {
-		log.Infof("not within grace [updatedOn %v] %v", marker.UpdatedOn.String(), time.Since(marker.UpdatedOn).String())
+	isWithinGrace, tilGrace := gracePeriod(marker, cfg.GracePeriod)
+	if isTerminalState(marker) || !isWithinGrace {
```

Comment:
something doesn't add up to me here, gracePeriod treats marker as always not nil, but isTerminalState supports market not being nil. Is this something we should align, is now marker able to pass down as nil?

Reply:
Marker itself is guaranteed to be non-nil by the `if marker == nil {return}` a few lines above this one.
Will clean up the check in `isTerminalState` and add godoc